department-of-veterans-affairs/caseflow
# Piedram/appeals 13365

Resolves [APPEALS-13367](https://vajira.max.gov/browse/APPEALS-13367), [APPEALS-13365](https://vajira.max.gov/browse/APPEALS-13365)

### Description

As a developer, I need to create the action's functionality for the "Send Initial Notification Letter" Task so that Contested Claim letter notifications can be tracked within Caseflow

### Acceptance Criteria:


##### The following occurs when the “Mark task as complete” task action is selected for the Send Initial Notification Letter task
- [ ] Modal appears for the user to indicate hold time from the following options: 45 Days and Custom Hold.
- [ ] User required to specify the number of days to place the task on hold (Min. value is 45 days and Maximum value is 364 days).
- [ ] When selecting Custom, appears a text area, and the button to send only is enabled when the values in the text area are between 46 and 364.

###### The following occurs when the user executes the “Mark task as complete” task action for the Send Initial Notification Letter Task
- [ ] “Send Initial Notification Letter” task is marked as complete.
- [ ] The task moves from the Currently Active Tasks section to Case Timeline and contains the following information
- Date of occurrence
- Assigned to: <organization>
- Assigned by: <user who completed action to assign the task to COB>
- Completed by: <user who completed Send Initial Notification Letter task action>
- Task Instructions: Hold Time: <user input in the hold time modal>
- [ ] “Post-Send Initial Notification Letter Holding Period” task is created and active and appears in the Currently Active 
- [ ] Tasks section and contains the following information
- Assigned to the COB organization
- [ ] Case now appears in the On Hold tab of the COB organization queue.
- [ ] If assigned to an individual, the person assigned to the task is maintained in the new task


### Testing Plan

1. Go to Caseflow login and sign in as `BVADWISE`
2. Navigate to Mail Intake 
3. Select the first option from the dropdown and click the button to continue 
4. Enter a veteran file number.
5. Fill out the form and select **direct review** for the review option type. Click the button at the bottom to continue.
6. On the add issues page, click the **add issues** button. Select a benefit type that contains a **contested claim** issue.
7. The intake should be completed.
8. Now change the user to a COB member.
9. Navigate to Queue and click on the dropdown in the right corner to switch views and select **Clerk of the Board team cases**
10. You have to see the appeal that creates in your previous steps.

![cc1](https://user-images.githubusercontent.com/110848569/217893923-588a8409-2d7e-42f8-a8f9-b116b1eb2809.PNG)

11. Select this appeal, and select "Mark as complete" in the dropdown.

![cc2](https://user-images.githubusercontent.com/110848569/217896740-639e3af4-0722-4fb8-8c11-37ab91d604f3.PNG)

13. Now you can select days or custom days between 45 and  364 days

![cc3](https://user-images.githubusercontent.com/110848569/217898046-be638d6d-3702-451f-a81d-31f42a363640.PNG)

in that case, we select a custom option with 125 days.

![cc4](https://user-images.githubusercontent.com/110848569/217898084-1884e312-6f4d-4d44-8a2a-b27e64787df2.PNG)

14. Click on "Mark as complete" button, and this Task passes from 'SendInitialNotificationLetter' to 'PostSendInitialNotificationLetterHoldingTask'

![cc5](https://user-images.githubusercontent.com/110848569/217898926-b88600ef-5945-4181-b354-86a0082704c3.PNG)

You can see this new  task on **Currently active tasks**

![cc6](https://user-images.githubusercontent.com/110848569/217899192-94ffe279-f3bb-40fe-82f2-f3dc48a5d285.PNG)

In the **Case Timeline** the previous task mark as complete too.

![cc7](https://user-images.githubusercontent.com/110848569/217899206-37d7779a-1b93-4fc9-999d-cb136836079d.PNG)
